### DISCUSSION TITLE
Map vs Filter: Performance

### DISCUSSION THREAD

**Discussion by bdkjones** [realm/realm-cocoa#7687] [Q&A]
I'd like to know how the performance of Realm's `Map` type compares to the performance of just using `filter` on a `List`. 

For example, suppose I have a model where I have 400,000 `FileItem` instances (a subclass of `Object`), each of which represents a file on disk. I need to quickly get the `FileItem` object for path `/some/file/path.txt`. Which route is more performant: 

1. Create a `Map<String: FileItem>` property, where the key is the path of each file?

2. Create a `List<FileItem>` property and then just use `.where({ $0.path == somePath })`?

# General Observation

There's often very little documentation about the best performance path in Realm. There's a *few* tricks, like not using `()` on the declaration of collections, and structuring filter closures so that the most performant checks come first (e.g. numbers before strings), but there's lot of situations like the above where I wonder how things work under the hood.

Where the Swift SDK hits Realm Core, is the `Map` object just a convenient "wrapper" around the same database queries that `filter` will generate? Or is `Map`, like `Dictionary`, actually an optimized data structure for random access by key? 

I wish the docs would offer more information about performance and how to maximize it. It's hard to implement a complex app multiple ways to discover performance by testing.

**Comment by dmorgereth** [comment]
What do you mean by "not using () on the declaration of collections"?
